src/wp-includes/media.php

```diff
@@ -1927,7 +1927,24 @@ function wp_img_tag_add_loading_attr( $image, $context ) {
  * @return string Converted `img` tag with `decoding=async` attribute added.
  */
 function wp_img_tag_add_decoding_async_attr( $image ) {
-	return str_replace( '<img ', '<img decoding="async" ', $image );
+	/**
+	 * Filters the `decoding` attribute value to add to an image. Default `async`.
+	 *
+	 * Returning `false` or an empty string will not add the attribute.
+	 *
+	 * @since 6.0.0
+	 *
+	 * @param string|bool $value   The `decoding` attribute value. Returning a falsey value will result in
+	 *                             the attribute being omitted for the image.
+	 * @param string      $image   The HTML `img` tag to be filtered.
+	 * @param string      $context Additional context about how the function was called or where the img tag is.
+	 */
+	$value = apply_filters( 'wp_img_tag_add_decoding_attr', 'async', $image, $context );
```

Comment:
Looks like `$context` is missing as an argument to this function as at this point is not defined. 

Additionally to that would be great to pass the $attachment_id similarly to:

- https://github.com/WordPress/wordpress-develop/blob/0edb1a7122b3db868a69b3c9119946281a16fdbc/src/wp-includes/media.php#L1969-L1981

Reply:
> Looks like $context is missing as an argument to this function as at this point is not defined.

this is fixed now, right?

> Additionally to that would be great to pass the $attachment_id similarly

good point, I'll add that

Reply:
> this is fixed now, right?

Right I was looking to an older diff :+1: